media/js/firefox/whatsnew/whatsnew-98-turningred-en.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+(function () {
+    'use strict';
+
+    // https://davidwalsh.name/javascript-debounce-function
+    function debounce(func, wait, immediate) {
+        var timeout;
+        return function () {
+            var context = this,
+                args = arguments;
+            var later = function () {
+                timeout = null;
+                if (!immediate) {
+                    func.apply(context, args);
+                }
+            };
+            var callNow = immediate && !timeout;
+            clearTimeout(timeout);
+            timeout = setTimeout(later, wait);
+            if (callNow) {
+                func.apply(context, args);
+            }
+        };
+    }
+
+    function isInViewport(element) {
+        var bounding = element.getBoundingClientRect();
+        var elementHeight = element.offsetHeight;
+        var elementWidth = element.offsetWidth;
```

Comment:
**nit:** remove width checks, I don't think we need to worry about right and left because it's a vertical list

Reply:
Good catch. This was swiped from somewhere else, but in this case, yeah, we only care about top and bottom so we're setting variables that will never be used.